django_modern_rest/serialization.py

```diff
@@ -124,3 +126,177 @@ def optimize_endpoint(cls, metadata: 'EndpointMetadata') -> None:
             metadata: Endpoint metadata to optimize.
 
         """
+
+
+@dataclass(slots=True, frozen=True)
+class _ComponentSpec:
+    name: str
+    data_getter: Any
+    model: Any
+    strict: bool
+
+
+class SerializerContext:  # noqa: WPS214
+    """Parse and bind request components for a controller.
+
+    This context collects raw data for all registered components, validates
+    the combined payload in a single call using a cached TypedDict model,
+    and then binds the parsed values back to the controller.
+    """
+
+    __slots__ = ('_combined_model', '_names', '_serializer', '_specs')
+
+    def __init__(
+        self,
+        specs: tuple[_ComponentSpec, ...],
+        serializer: type[BaseSerializer],
+        combined_model: Any,
+        names: tuple[str, ...],
+    ) -> None:
+        """Initialize the context with a compiled plan and combined model."""
+        self._specs = specs
+        self._serializer = serializer
+        self._combined_model = combined_model
+        self._names = names
+
+    def parse_and_bind(
+        self,
+        controller: Any,
+        request: HttpRequest,
+        *args: Any,
+        **kwargs: Any,
+    ) -> None:
+        """Collect, validate, and bind component data to the controller."""
+        context = self._collect_context(controller, request, *args, **kwargs)
+        validated = self._validate_context(context)
+        self._bind_parsed(controller, validated)
+
+    @classmethod
+    def for_controller(
```

Comment:
Sorry, not clear why we would need this method. Controller already has `_serializer_context` prop. Just use it.